shell.nix

```diff
@@ -1,15 +1,10 @@
+{ system ? builtins.currentSystem}:
 let
-  pkgs = import ./nix {};
+  pkgs = import ./nix { inherit system; };
 in
 pkgs.mkShell {
-  buildInputs = [
-    pkgs.stack
-    pkgs.niv
-    # hie can't find hspec-discover via stack
-    pkgs.haskellPackages.hspec-discover
-    pkgs.haskellPackages.releaser
-    pkgs.haskellPackages.hkgr
-  ];
+  buildInputs = (with pkgs; [ stack niv ]) ++
```

Comment:
We generally want to avoid `with`: https://nix.dev/anti-patterns/language#with-attrset-expression